tests/integration/test_index.py

```diff
@@ -14,6 +14,72 @@
 
 from gqlalchemy.models import MemgraphIndex
 from gqlalchemy import Field, Node
+import pytest
+from gqlalchemy.exceptions import GQLAlchemyDatabaseMissingInNodeClassError
+
+
+def test_index_label(memgraph):
+    class User(Node, index=True, db=memgraph):
+        id: str
+
+    actual_index = memgraph.get_indexes()
+
+    assert set(actual_index) == {MemgraphIndex("User")}
+
+
+def test_missing_db_in_node_class(memgraph):
+    with pytest.raises(GQLAlchemyDatabaseMissingInNodeClassError):
+
+        class User(Node, index=True):
+            id: str
+
+
+def test_db_in_node_class(memgraph):
+    class User(Node, db=memgraph):
+        id: str = Field(index=True)
+
+    actual_index = memgraph.get_indexes()
+    assert set(actual_index) == {MemgraphIndex("User", "id")}
+
+
+def test_db_in_node_and_property(memgraph):
+    class User(Node, db=memgraph):
+        id: str = Field(index=True, db=memgraph)
+
+    actual_index = memgraph.get_indexes()
+    assert set(actual_index) == {MemgraphIndex("User", "id")}
+
+
+def test_index_on_label_and_property(memgraph):
+    class User(Node, index=True, db=memgraph):
+        id: str = Field(index=True, db=memgraph)
+
+    actual_index = memgraph.get_indexes()
+    assert set(actual_index) == {MemgraphIndex("User", "id"), MemgraphIndex("User")}
+
+
+def test_false_index_in_node_class(memgraph):
+    class User(Node, index=False, db=memgraph):
+        id: str
+
+    actual_index = memgraph.get_indexes()
+    assert set(actual_index) == set()
+
+
+def test_false_index_no_db_in_node_class(memgraph):
+    class User(Node, index=False):
+        id: str
+
+    actual_index = memgraph.get_indexes()
+    assert set(actual_index) == set()
+
+
+def test_false_index_with_db_in_node_class(memgraph):
+    class User(Node, index=False, db=memgraph):
```

Comment:
Can we promote these in-test defined classes to fixtures, and use them somehow as parameters for the tests?